ground/src/main/java/com/google/android/ground/persistence/remote/RemoteDataEvent.kt

```diff
@@ -15,20 +15,15 @@
  */
 package com.google.android.ground.persistence.remote
 
-import com.google.android.ground.rx.ValueOrError
-
 /**
  * An event returned by the remote data store indicating either an entity was successfully loaded,
  * or that it was modified or removed in the remote data store.
  *
  * @param <T> the type of entity being loaded, modified, or removed.
- *
- * TODO: Replace with kotlin.Result once migration to Kotlin is complete. For more info, see this
- *  https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/
  */
 class RemoteDataEvent<T> private constructor(
-    val entityId: String, val eventType: EventType, entity: T?, error: Throwable?
-) : ValueOrError<T>(entity, error) {
+    val eventType: EventType, val entityId: Result<String?>, val entity: Result<T?>
```

Comment:
The way this is modeled would imply that retrieving `entityId` and `entity` can succeed or fail independently. Perhaps we should instead model this as `Result<Pair<String, T>>` so that they're bundled together?